server/stream.go

```diff
@@ -666,3 +670,33 @@ func (mset *Stream) ackMsg(obs *Consumer, seq uint64) {
 		}
 	}
 }
+
+func (mset *Stream) sendJSError(c *client, caller *client, event JSError) {
+	if event.err != nil {
+		c.Errorf(event.LogLine())
+	} else {
+		c.Warnf(event.LogLine())
+	}
+
+	if event.ShouldRateLimit() && !errorEventLimit.Allow() {
+		return
+	}
+
+	e := event.Advisory()
+
+	mset.mu.Lock()
+	e.Stream = mset.config.Name
+	c.mu.Lock()
+	e.Server = mset.client.srv.Name()
```

Comment:
I assume that if you lock `c.mu.Lock()` it is because `c==mset.client`, right? But you can't call srv.Name() inside the client lock, that could create a lock inversion (Name() is locking *Server).
So you should write something like that instead:
```
e := event.Advisory()

mset.mu.RLock()
e.Stream = mset.config.Name
mset.mu.RUnlock()

c.mu.Lock()
srv := c.srv
c.mu.Unlock()
if srv != nil {
    e.Server = srv.Name()
}
```

.. and same comment than in Account.sendJSError() due to use of public API Server.Name() that requires server lock.